File: src/lib/distribution.ts
/** Build-time distribution settings; ordinary builds retain upstream behavior. */
export const IS_AIGOCODE_COMPAT = import.meta.env.VITE_AIGOCODE_COMPAT === "1";

export const RELEASES_URL = IS_AIGOCODE_COMPAT
  ? import.meta.env.VITE_AIGOCODE_RELEASES_URL ||
    "https://github.com/jupiterchu/cc-switch/releases"
  : "https://github.com/farion1231/cc-switch/releases";

File: src-tauri/src/distribution.rs
/// Compatibility builds use manual releases so upstream updates cannot remove the patch.
pub(crate) fn is_aigocode_compat() -> bool {
    option_env!("AIGOCODE_COMPAT_BUILD") == Some("1")
}

pub(crate) fn releases_url() -> &'static str {
    if is_aigocode_compat() {
        option_env!("AIGOCODE_RELEASES_URL")
            .filter(|url| !url.is_empty())
            .unwrap_or("https://github.com/jupiterchu/cc-switch/releases")
    } else {
        "https://github.com/farion1231/cc-switch/releases/latest"
    }
}
